### PR TITLE
Create sourceset documentable filter for javadoc

### PR DESCRIPTION
**Pull request by MarcinAman**
PR is waiting for #1177 to be finished due to merger changes

### CODE REVIEW

File: plugins/javadoc/src/main/kotlin/javadoc/transformers/documentables/JavadocDocumentableSourceSetFilterTransformer.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package javadoc.transformers.documentables
+
+import org.jetbrains.dokka.Platform
+import org.jetbrains.dokka.model.DModule
+import org.jetbrains.dokka.model.Documentable
+import org.jetbrains.dokka.transformers.documentation.PreMergeDocumentableTransformer
+
+class JavadocDocumentableSourceSetFilterTransformer : PreMergeDocumentableTransformer {
+    private val List<Documentable>.jvmSourceSets
+        get() = flatMap { it.sourceSets }.filter { it.analysisPlatform == Platform.jvm }.distinct()
```
Comment (kamildoleglo) [inline plugins/javadoc/src/main/kotlin/javadoc/transformers/documentables/JavadocDocumentableSourceSetFilterTransformer.kt:10]: I'd personally also leave the `common` source sets that those jvm source sets depend upon

Comment (MarcinAman) [reply plugins/javadoc/src/main/kotlin/javadoc/transformers/documentables/JavadocDocumentableSourceSetFilterTransformer.kt:10]: uh ok, that means that i need to also figure out how to merge those sourcesets to ensure javadoc only works on 1 of them

Comment (kamildoleglo) [reply plugins/javadoc/src/main/kotlin/javadoc/transformers/documentables/JavadocDocumentableSourceSetFilterTransformer.kt:10]: Why, can't we just use the Documentable merger as always? 

Also, I'd log an info that some source sets were dropped

Comment (kamildoleglo) [reply plugins/javadoc/src/main/kotlin/javadoc/transformers/documentables/JavadocDocumentableSourceSetFilterTransformer.kt:10]: Bump

**Pull request closed without merging**